sound/soc/sof/sof-audio.c

```diff
@@ -343,6 +343,9 @@ sof_setup_or_free_widgets_in_order(struct snd_sof_dev *sdev, struct snd_soc_dapm
 	struct snd_soc_dapm_widget *widget;
 	int ret, i;
 
+	if (!list)
+		return 0;
+
```

Comment:
it's either
a) a false alarm - this is already tested in a macro
b) a real problem, the same code is used in soc-dapm.c

````c
	wlist = dapm_kcontrol_get_wlist(update->kcontrol);

	for_each_dapm_widgets(wlist, wi, w) {

#define for_each_dapm_widgets(list, i, widget)				\
	for ((i) = 0;							\
	     (i) < list->num_widgets && (widget = list->widgets[i]);	\
	     (i)++)

````

Reply:
this is definitely a false alarm. the csame check is there in the caller function too.

Reply:
let's remove it then and mark the issue as N/A

Reply:
dropped now